qiskit_experiments/library/characterization/analysis/ramsey_xy_analysis.py

```diff
@@ -412,16 +424,93 @@ def _default_options(cls):
-                "Yneg": {"series": "Y", "direction": "neg"},
+                "Fpos": {"direction": "pos"},
+                "Fneg": {"direction": "neg"},
             },
             plotter=ramsey_plotter,
         )
 
         return options
 
+    def _format_data(
+        self,
+        curve_data: curve.ScatterTable,
+    ) -> curve.ScatterTable:
+        columns = list(curve_data.columns)
+
+        i = 0
+        # Reassign model class.
+        # These data are actually P1 value of Ramsey X, Y quadrature,
+        # but the data is automatically classified based on the fit models,
+        # which are for the synthetic phase data in this analysis.
+        # This yields false classification.
+        for series in ("X", "Y"):
+            for direction in ("pos", "neg"):
+                rows = (curve_data.series == series) & (curve_data.direction == direction)
+                curve_data.loc[rows, "model_name"] = f"{series}{direction}"
+                # Assign random model id which will be invalidated later.
+                # Note that super()._format_data relies on the model id rather than
+                # model name (i.e. integer sort is faster in pandas) to
+                # average over the same x value in the same group.
+                curve_data.loc[rows, "model_id"] = i
+                i += 1
+        formatted_data = super()._format_data(curve_data)
+        # Invalidate model_id because these are index of raw Ramsey curves.
+        # Fit models are defined based on the phase data.
+        formatted_data.model_id = pd.NA
+
+        # Convert to phase format with unwrapper
+        raw_ramsey = formatted_data.filter(like="formatted", axis="index")
+        phase_data = np.empty((0, len(columns)))
+        y_mean = raw_ramsey.yval.mean()
+        grouped = raw_ramsey.groupby(["direction", "series"])
+
+        for m_id, direction in enumerate(("pos", "neg")):
+            x_quadrature = grouped.get_group((direction, "X"))
+            y_quadrature = grouped.get_group((direction, "Y"))
+            if not np.array_equal(x_quadrature.xval, y_quadrature.xval):
+                raise ValueError(
+                    "Amplitude values of X and Y quadrature are different. "
+                    "Same values must be used."
+                )
+            x_uarray = unp.uarray(x_quadrature.yval, x_quadrature.yerr)
+            y_uarray = unp.uarray(y_quadrature.yval, y_quadrature.yerr)
+
+            amplitudes = x_quadrature.xval.to_numpy()
+
+            # pylint: disable=no-member
+            phase = unp.arctan2(y_uarray - y_mean, x_uarray - y_mean)
+            phase_n = unp.nominal_values(phase)
+            phase_s = unp.std_devs(phase)
+
+            # Unwrap phase
+            # We assume a smooth slope and correct 2pi phase jump to minimize the change of the slope.
+            if amplitudes[0] < 0:
+                # Flip array order because this is array is negative increments.
```

Comment:
Something like this?

```suggestion
                # Flip array order because this is array runs large negative numbers and large phase values towards zero amplitude and smaller phase values. we want to preserve the phase value closest to zero amplitude.
```